ccm_web/server/src/main.ts

```diff
@@ -68,6 +69,7 @@ async function bootstrap (): Promise<void> {
       }
     })
   )
+  app.useGlobalInterceptors(new CacheControlToHeaderInterceptor())
```

Comment:
I Felt adding it globally is a best way than adding a Decorator at the API level  https://docs.nestjs.com/controllers#headers

Reply:
My main question is whether adding it globally here will affect the static routes. My guess is no currently, but it may once we make a change like in #316, because the static serving will be installed at the module level, not globally. Hm...

Reply:
Then I can add this interceptor at the API level, if you feel it is going to disrupt the static routes with #316. Let me pull your change from PR  #316 and combine mine and see if anything changes.

As of now the static assets like the `main-sdsd.js` file have always caching on it.

Reply:
@ssciolla answers from these https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/321#discussion_r797989387 will help define it.

Reply:
Looks like the global interceptor doesn't really affect the static. It may be slightly clearer to install the interceptor in `app.module.ts`, using the latter pattern here: https://docs.nestjs.com/interceptors#binding-interceptors. I'll leave that up to you. I like that we wouldn't have to import any Nest `Injectable`s except `app.module.ts` in `main.ts`, and that it would automatically be included in any test modules if we ever get to building out the end-to-end tests.

Reply:
Looks like the global interceptor doesn't really affect the static. It may be slightly clearer to install the interceptor in `app.module.ts`, using the latter pattern here: https://docs.nestjs.com/interceptors#binding-interceptors. I'll leave that up to you. I like that we wouldn't have to import any Nest `Injectable`s except `app.module.ts` in `main.ts`, and that it would automatically be included in any test modules if we ever get to building out the end-to-end tests.